macros/models/_sector/dex/enrich_dex_trades.sql

```diff
@@ -47,10 +47,17 @@ WITH base_trades as (
         , base_trades.token_sold_amount_raw / power(10, erc20_sold.decimals) AS token_sold_amount
         , base_trades.token_bought_amount_raw
         , base_trades.token_sold_amount_raw
-        , coalesce(
-                base_trades.token_bought_amount_raw / power(10, erc20_bought.decimals) * p_bought.price,
-                base_trades.token_sold_amount_raw / power(10, erc20_sold.decimals) * p_sold.price
-            ) AS amount_usd
+        , CASE
+                WHEN stability_bought.rank IS NOT NULL AND (stability_sold.rank IS NULL OR stability_bought.rank < stability_sold.rank) THEN
+                    base_trades.token_bought_amount_raw / power(10, erc20_bought.decimals) * p_bought.price
+                WHEN stability_sold.rank IS NOT NULL AND (stability_bought.rank IS NULL OR stability_sold.rank < stability_bought.rank) THEN
+                    base_trades.token_sold_amount_raw / power(10, erc20_sold.decimals) * p_sold.price
+                ELSE
+                    coalesce(
+                        base_trades.token_bought_amount_raw / power(10, erc20_bought.decimals) * p_bought.price,
+                        base_trades.token_sold_amount_raw / power(10, erc20_sold.decimals) * p_sold.price
+                    )
+            END AS amount_usd
```

Comment:
let's make this a macro for sure. we should initiate for dex, then reuse across all sectors for consistency as needed

Reply:
i think what we can do is similar to the `add_tx_from_to` macro, where we group all logic in a CTE, then pass CTE into macro to join to new prices spell below and write this type of logic

if we can't nest macros like that, then i'd suggest we remove `amount_usd` from this macro altogether, and then try to do directly in final dex spell maybe? hard to say exactly without being hands-on with it here but just giving some ideas

Reply:
Since amount_usd calculation involves `erc20.decimals`, `token.price`, the `add_amount_usd` will be very similar to `enrich_dex_trades`

Reply:
I just drafted the `add_amount_usd` macro.

I am thinking if I should hard-coded the column names or introduce them as parameters in the signature.